src/Tickets/Commerce/Gateways/Square/Hooks.php

```diff
@@ -170,6 +171,18 @@ public function process_async_webhook( int $order_id, int $retry = 0 ): void {
 				continue;
 			}
 
+			$event_id = $pending_webhook['metadata']['event_id'] ?? '';
+
+			if ( $event_id ) {
+				Webhook_Model::update(
```

Comment:
Could this throw? Should a throw here be handled specifically with some logging?